src/pkg/repository/data_providers.go

```diff
@@ -30,6 +36,12 @@ type DataProviderConnector interface {
 		ctx context.Context,
 		pst path.ServiceType,
 	) error
+	// DataProvider retrieves the data provider.
+	DataProvider() DataProvider
+}
+
+func (r *repository) DataProvider() DataProvider {
+	return r.Provider
```

Comment:
is this still used? If not, please remove it to help keep interfaces smaller

Reply:
still used, yes.